generator/ServiceClientGeneratorLib/Generators/Marshallers/RestXmlRequestMarshaller.tt

```diff
@@ -736,4 +750,39 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
 <#=new string(' ', (level) * 4)#>					}
 <#+
 	}
+#>
+<#+
+	// xsi is a common prefix in attributes, but it cannot be included in the local name itself as the xmlWriter class will throw an exception. 
+	// Some services may model the xmlName, with the xsi prefix included like it is here https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/restXmlWithNamespace/main.smithy#L147
+	// In this case, we add xsi to the prefix argument of WriteAttributeString and extract the local name.
+	protected void WriteXmlAttributeString(int level, Member member, string variableName,  bool isPayload, Operation operation = null)
+	{
+		var prefix = "";
+		var localName = "";
+		string ns = member.XmlNamespace;
+		if (member.MarshallName.StartsWith("xsi"))
```

Comment:
Should this be generic prefix parsing instead?

Reply:
Just for some context 

Normally a service would model it like this where they tell us the prefix. In that case we use the prefix provided https://github.com/aws/aws-sdk-net/blob/806efa402e867e444e309960fbb688d6e1bda966/generator/TestServiceModels/restxmlwithnamespace-tests-client/rest-xml-protocol-namespace-2019-12-16.normal.json#L100

But in some cases the service will actually attach `xsi` to the locationName itself like this https://github.com/aws/aws-sdk-net/blob/806efa402e867e444e309960fbb688d6e1bda966/generator/TestServiceModels/restxmlwithnamespace-tests-client/rest-xml-protocol-namespace-2019-12-16.normal.json#L60

Although no actual service does this, the protocol tests want to make sure we can handle that. The only other valid prefix from what I've seen is `xsd` (other than `xsi`), but I don't see a problem with just assuming everything before the first `:` is a prefix, since we can't unmarshall an attribute with a `:` in it anyways. Will make it more generic.